### ISSUE TITLE
PyTables replacement for numpy.savez

### ISSUE THREAD

**Issue by kyleabeauchamp** [PyTables/PyTables#185]
In many situations, I find myself wanting to quickly load large numpy arrays from disk.  However, the existing numpy tools could use some improvement in terms of both speed and file size.  I propose that PyTables include a replacement for numpy's binary save / load feature (e.g. np. savez).  

Our (@rmcgibbo) proposal is to create quick_and_dirty IO functions in PyTables called saveh() and loadh().   These functions would function almost identically to numpy.savez() and numpy.load(), but would use PyTables underneath.  

We have already implemented this feature in our own codebase (https://github.com/SimTk/msmbuilder/) and can prepare a pull request if people find this useful.  As an example, below I compare the performance of our PyTables based approach to np.savez.  The PyTables based approach shows reduced file size (10-20%) and improved read performance (~160%).  Although the relative write performance of Numpy versus PyTables does depend on the details of data, compression, and chunk,  in almost all cases I see the PyTables approach bringing dramatic improvements in read performance.  

```
import numpy as np
from msmbuilder import io

n = 150000000
x = normal(size=(n))

%time np.savez("test.npz",x)  # Wall time: 10.58 s
%time np.savez_compressed("./test_compressed.npz",x)  # Wall time: 105.00 s
%time io.saveh("./test.hdf",x)  # Wall time: 5.92 s

%time x = np.load("./test.npz")["arr_0"]  # Wall time: 3.91 s
%time x = np.load("./test_compressed.npz")["arr_0"]  # Wall time: 16.88 s
%time x = io.loadh("./test.hdf")["arr_0"]  # Wall time: 2.24 s
```

To summarize, here are the key advantages:
1.  Expected read performance > 100% better than existing solutions.
2.  Fast compression, small file sizes
3.  Encourage people to use PyTables for day-to-day data storage.
